### PR TITLE
dts: arm: nordic: Remove label property from devicetrees

### PR DESCRIPTION
**Pull request by galak**
Label properties are not required.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

File: dts/arm/nordic/nrf51822.dtsi [hunk 1]
```diff
@@ -92,7 +89,6 @@
 			clock-frequency = <I2C_BITRATE_STANDARD>;
 			interrupts = <3 NRF_DEFAULT_IRQ_PRIORITY>;
 			status = "disabled";
-			label = "I2C_0";
```
Comment (anangl) [inline dts/arm/nordic/nrf51822.dtsi:95]: Don't we need to still keep some of these labels for shell? At least `I2C_*` ones seem to be needed: https://github.com/zephyrproject-rtos/zephyr/blob/3b5590a18790ea8f9159eadbc38f8a514b342543/drivers/i2c/i2c_shell.c#L280

Comment (galak) [reply dts/arm/nordic/nrf51822.dtsi:95]: So looking at this, I think we can rework this based on infrastructure we have for userspace.  We can find a list of devices that support the API of interest.

Comment (galak) [reply dts/arm/nordic/nrf51822.dtsi:95]: Here's the starts of a change that implements a better filter/lookup mechanism:
https://github.com/zephyrproject-rtos/zephyr/pull/48381

**Comment by gmarull** [comment]
looks like MCUboot will need to be moved away from `device_get_binding` as well.

**Comment by galak** [comment]
> looks like MCUboot will need to be moved away from `device_get_binding` as well.

already done.

**Comment by gmarull** [comment]
compliance hotfix: https://github.com/zephyrproject-rtos/zephyr/pull/48562

**Review by gmarull — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by anangl — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**